index.d.ts

```diff
@@ -0,0 +1,67 @@
+type Availability = {
+    /**
+     * ISO 8601 string
+     */
+    start: string
+    /**
+     * ISO 8601 string
+     */
+    end: string
+}
+
+declare class AvailabilitySchedule {
+    constructor(
+        /**
+         * Start date of the schedule as an ISO 8601 string
+         */
+        startDate: string,
+        /**
+         * End date of the schedule as an ISO 8601 string
+         */
+        endDate: string
+    )
+
+    /**
+     * @param startDate ISO 8601 string
+     * @param endDate ISO 8601 string
+     */
+    addAvailability(startDate: string, endDate: string): void
+
+    /**
+     * Start and end dates may be earlier than the start date passed to the constructor.
+     * @param startDate ISO 8601 string
+     * @param endDate ISO 8601 string
+     * @param repeatWeekdays Array of integers that indicate the weekdays on which the availability is repeated. 1 = Mon, 7 = Sun (in the same time zone as startDate).
+     */
+    addWeeklyRecurringAvailability(
+        startDate: string,
+        endDate: string,
+        repeatWeekdays: number[]
+    ): void
+
+    /**
+     * Use this to remove a time range from any existing availabilities.
+     * For example, call this for scheduled appointments or meetings.
+     * @param startDate ISO 8601 string
+     * @param endDate ISO 8601 string
+     */
+    removeAvailability(startDate: string, endDate: string): void
+
+    /**
+     * Returns all availabilities in chronological order.
+     * @param timezone Defaults to '+0000'
+     */
+    getAvailabilities(timezone?: string): Availability[]
+
+    /**
+     * Returns true if the given time range falls within any availability.
+     * @param startDate ISO 8601 string
+     * @param endDate ISO 8601 string
+     */
+    isAvailable(startDate: string, endDate: string): boolean
+}
+
+declare module 'availability-schedule' {
+    const AvailabilitySchedule: AvailabilitySchedule
```

Comment:
This line should be removed. TS then correctly picks up the class export when imported with:
```ts
import AvailabilitySchedule = require('availability-schedule');
```
(which I'll add to the readme). See https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

Reply:
This doesn't work with ES modules: `TS1202: Import assignment cannot be used when targeting ECMAScript modules`

Reply:
How are you importing it? Does this require more changes or all good?

Reply:
I think all good

Reply:
`import AvailabilitySchedule from 'availability-schedule'`

Reply:
Oh nice, I didn't think that works. I'll add a note for the error you got and that the regular "import from" statement works when targeting ES modules.